File: src/main/repositories/config-repository.ts
import { existsSync, readFileSync, writeFileSync } from "node:fs";
import { Config } from "../models/config";

export class ConfigRepository {
  constructor(private readonly configPath: string) {}

  load(): Config | null {
    if (!existsSync(this.configPath)) {
      this.save(new Config());
      return null;
    }

    try {
      const data = JSON.parse(readFileSync(this.configPath, "utf-8"));
      const config = Config.fromDict(data);
      config.normalizePaths();
      return config;
    } catch {
      this.save(new Config());
      return null;
    }
  }

  save(config: Config): void {
    writeFileSync(
      this.configPath,
      JSON.stringify(config.toDict(), null, 4),
      "utf-8"
    );
  }
}

File: .mewtator/app/infrastructure/config_repository.py
import json
import os
from typing import Optional
from app.core.models.config import Config


class ConfigRepository:
    def __init__(self, config_path: str = "config.json"):
        self.config_path = config_path
    
    def load(self) -> Optional[Config]:
        if not os.path.exists(self.config_path):
            self.save(Config())
            return None
        
        try:
            with open(self.config_path, "r", encoding="utf-8") as f:
                data = json.load(f)
                config = Config.from_dict(data)
                config.normalize_paths()
                return config
        except (json.JSONDecodeError, IOError):
            self.save(Config())
            return None
    
    def save(self, config: Config):
        with open(self.config_path, "w", encoding="utf-8") as f:
            json.dump(config.to_dict(), f, indent=4)
